### PR TITLE
Add ability to auto opt-in for meeting requests to subscriptions

### PR DESCRIPTION
**Pull request by MinaAtYelp**
[[BEANS-94](https://jira.yelpcorp.com/browse/BEANS-94)] Add ability to auto opt-in for meeting requests to subscriptions 

This also fixes [BEANS-72](https://jira.yelpcorp.com/browse/BEANS-72)

### Backend changes
- Added `default_auto_opt_in` field to `MeetingSubscription` model, represents the default value for whether auto opt in for meeting requests is enabled for a particular `MeetingSubscription` (default: `False`)
- Added `auto_opt_in` field to `UserSubscriptionPreferences` model, represents whether auto opt in for meeting requests is enabled for a particular `UserSubscriptionPreferences`  (default: `False`) 
- Updated `send_batch_weekly_opt_in_email()` to exclude `UserSubscriptionPreferences` with `auto_opt_in` == `True`
- Added a cron job to generate meeting requests for `UserSubscriptionPreferences` with `auto_opt_in` == `True`
    - Runs weekly 
       - after the meetings spec for the current week have been generated (`/tasks/generate_meeting_specs_for_week`)
       - before running match batch (`/tasks/match_employees`)   
 

### UI changes

- `SubscriptionList.js` (accessed via http://yelp-beans.appspot.com/admin/subscriptions/<subscription_id>) 
   - Added ability to display and update `default_auto_opt_in` of each subscription 
- `UserPreferenceForm.js` (accessed via http://yelp-beans.appspot.com/) 
   - Swapped `checkbox` denoting whether user is subscribed to a meeting date for each subscription with a `toggle`
   - Added a second toggle for enabling/disabling Auto opt in for meeting requests (`auto_opt_in`) of a meeting date for each subscription    
   - General style/css updates 

https://github.com/Yelp/beans/assets/136340867/b78d22b4-75ac-49a9-8010-1c7cfda631f0

Side note: `http://yelp-beans.appspot.com/admin/subscriptions/` wasn't working on dev due to a CORS error which I believe is related to codespaces port forwarding setup

https://github.com/Yelp/beans/assets/136340867/b90a5f9b-8244-454b-a2fe-fb2b7e96d9e9

### CODE REVIEW

**Comment by MinaAtYelp** [comment]
UI after pushing fix for aligning Bio card with preferences  [da41ded](https://github.com/Yelp/beans/pull/256/commits/da41ded9234f8f408516c9dfd1841241fa6760da)
Screenshot 2023-09-06 at 1 41 39 AM

**Review by ymilki — COMMENTED** [review]
This is great! Thanks for bringing this over the finish line. We've had previous work on this in #158 by @twof as well that didn't quite make it to completion. Good to see there are lots of tests too.

I'm not so sure about the UI changes, but we can iterate on those later. Maybe someone with a keen eye on design will be motivated to help once this reaches prod.

Note: We will need to do the schema changes out-of-band of the PR.

**Comment by zachche** [comment]
this is awesome! thanks for adding so much; i can't wait to see what it'll do for beans adoption :) i'm not super familiar with the code so i'll leave that up to @kdeal or someone else more familiar. 

in terms of usability, i don't think most people even know about the meeting opt in. it might be more clear if we update the copy to describe what's happening (also open to other suggestions):
- (Subscription) `Enabled` --> `Subscribe`
- Opt in button
  - When enabled:
    - `Auto opt-in` --> `Automatically opt-in to weekly matches`
  - When disabled:
    - `Manually opt-in each week via email from beans@yelp.com`

File: api/yelp_beans/routes/tasks.py [hunk 1]
```diff
@@ -104,3 +108,40 @@ def clean_user_subscriptions():
...
+@tasks.route("/generate_meeting_requests_for_auto_opt_in_preferences", methods=["GET"])
+def generate_meeting_requests_for_auto_opt_in_preferences():
+    specs = get_specs_for_current_week()
+
+    logging.info("All specs: ")
+    logging.info(specs)
+
+    for spec in specs:
+        spec_timezone = timezone(spec.meeting_subscription.timezone)
+        spec_dt = spec.datetime.replace(tzinfo=utc).astimezone(spec_timezone)
```
Comment (kdeal) [inline api/yelp_beans/routes/tasks.py:122]: Should we be using the `get_meeting_datetime` function [here](https://github.com/MinaAtYelp/beans/blob/da41ded9234f8f408516c9dfd1841241fa6760da/api/yelp_beans/logic/meeting_spec.py#L65)?
```suggestion
       spec_dt = get_meeting_datetime(spec)
```

**Review by kdeal — APPROVED** [review]
Looks good, thanks for implementing this feature!

File: api/yelp_beans/routes/tasks.py [hunk 2]
```diff
@@ -104,3 +108,39 @@ def clean_user_subscriptions():
...
+
+@tasks.route("/generate_meeting_requests_for_auto_opt_in_preferences", methods=["GET"])
+def generate_meeting_requests_for_auto_opt_in_preferences():
+    specs = get_specs_for_current_week()
+
+    logging.info("All specs: ")
+    logging.info(specs)
+
+    for spec in specs:
+       spec_dt = get_meeting_datetime(spec)
```
Comment (kdeal) [inline api/yelp_beans/routes/tasks.py:121]: My previous suggestion was off by one space I think.
```suggestion
        spec_dt = get_meeting_datetime(spec)
```

File: api/yelp_beans/routes/tasks.py [hunk 3]
```diff
@@ -104,3 +108,39 @@ def clean_user_subscriptions():
...
+        user_sub_preferences_with_auto_opt_in = (
+            UserSubscriptionPreferences.query.filter(UserSubscriptionPreferences.subscription_id == spec.meeting_subscription_id)
+            .filter(UserSubscriptionPreferences.auto_opt_in.is_(True))
+            .all()
+        )
+
+        logging.info(f"All UserSubscriptionPreferences with auto_opt_in == True for spec = {spec}:")
+        logging.info(user_sub_preferences_with_auto_opt_in)
+        for user_preference in user_sub_preferences_with_auto_opt_in:
+            preference_dt = user_preference.preference.datetime.replace(tzinfo=utc).astimezone(spec_timezone)
```
Comment (kdeal) [inline api/yelp_beans/routes/tasks.py:132]: My first suggestion also removed this variable.
```suggestion
            preference_dt = user_preference.preference.datetime.replace(tzinfo=utc).astimezone(
                timezone(spec.meeting_subscription.timezone)
            )
```

**Pull request merged**